src/api/app/controllers/webui/status_messages_controller.rb

```diff
@@ -44,7 +56,7 @@ def destroy
       flash[:error] = "Could not delete status message: #{status_message.errors.full_messages.to_sentence}"
     end
 
-    redirect_to(controller: 'main', action: 'index')
+    redirect_back_or_to({ action: 'index' })
```

Comment:
This redirects to the referrer if it's there. The expected referrers are the homepage (`main#index`) or `Manage News`. If somehow the referrer is not there, we redirect to `Manage News`.